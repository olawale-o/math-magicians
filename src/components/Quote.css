.Quote {
  padding: 15px;
}