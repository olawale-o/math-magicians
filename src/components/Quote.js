import React from 'react';
import './Quote.css';

const quoteStyle = {
  width: '85%',
  margin: '10rem auto',
  fontSize: '3.5rem',
  fontWeight: 'light',
};
const Quote = () => (
  <div className="Quote">
    <p style={quoteStyle} data-testid="quote">
      Mathematics is not about numbers, equations, computations
      <br />
      or algorithms: it is about understanding. -William Paul Thurston
    </p>
  </div>
);

export default Quote;
