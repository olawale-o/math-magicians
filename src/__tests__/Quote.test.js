import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Quote from '../components/Quote';

describe('Home', () => {
  it('Renders Quote component correctly and create snapshot', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it has a header Element', () => {
    render(<Quote />);

    const quote = screen.getByTestId('quote');

    expect(quote).toBeTruthy();
    expect(quote).toBeInTheDocument();

    expect(quote).toHaveTextContent(
      /Mathematics is not about numbers, equations, computations/i,
    );
  });
});
