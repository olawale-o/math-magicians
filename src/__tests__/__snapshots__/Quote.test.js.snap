// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Quote component correctly and create snapshot 1`] = `
<div
  className="Quote"
>
  <p
    data-testid="quote"
    style={
      Object {
        "fontSize": "3.5rem",
        "fontWeight": "light",
        "margin": "10rem auto",
        "width": "85%",
      }
    }
  >
    Mathematics is not about numbers, equations, computations
    <br />
    or algorithms: it is about understanding. -William Paul Thurston
  </p>
</div>
`;
